utilidades/storage_file_utils.py

```
@@ -481,11 +514,19 @@ def drop_processed_duplicates(df: pd.DataFrame, dataset_type: str) -> pd.DataFra
                 df = df.drop_duplicates(subset=['datetime_utc', 'mercado', "tecnologia"], keep='last')
+            elif dataset_type == 'precios_i90':
+                df = df.drop_duplicates(subset=['datetime_utc', 'mercado', "up"], keep='last')
+            elif dataset_type == 'volumenes_omie':
+                df = df.drop_duplicates(subset=['datetime_utc', 'mercado', "uof"], keep='last')
+            elif dataset_type == 'volumenes_mic':
+                #we allow duplicates for mic market data, so we don't drop any duplicates
+                return df
             else:
                 df = df.drop_duplicates(subset=['datetime_utc', 'id_mercado', 'precio'], keep='last')
+        except KeyError as e:
+            raise KeyError(f"Missing required column for deduplication: {str(e)}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)